conf/machine/xenclient-common.conf

```diff
@@ -11,6 +11,7 @@ require conf/machine/include/tune-core2.inc
 # GCC to do that again until someone upgrades GHC6 or replace Haskell
 # components.
 # The trade off with -mstackrealign should be less than disabling SSE.
-TUNE_CCARGS += "-mstackrealign"
+
+TUNE_CCARGS_remove += "-mstackrealign"
```

Comment:
Is "-mstackrealign" in TUNE_CCARGS such that we need to remove, or should TUNE_CCARGS and comment be deleted?

Reply:
I felt leaving this in was a defensive choice. Grepping over the layers shows that no recipes are setting mstackrealign, but it's possible one might in the future for some unknown reason.

If you think that's low enough risk we should be able to drop it altogether.

Reply:
It seems unlikely.  But, if you want to be defensive, you should update the comment to make it clear why it is present.  It would be nice if bitbake/oe has a "build-time assert" for checking something like this.

Reply:
It seems more likely that different alignment would be enforced using `-mpreferred-stack-boundary`. `-mstackrealign` will just allow mixing legacy code (4-byte aligned stack) with modern ones (16-bytes). Unless GHC is doing something funny, `-mstackrealign` in 64bit most likely has no effect?

Newer GHC versions probably have `force_align_arg_pointer` for unaligned stack calling to libc functions, if such things are still around.